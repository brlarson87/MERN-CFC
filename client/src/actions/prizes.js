import axios from "axios";
import {
  PRIZES_LOADED,
  CLEAR_PRIZES,
  PRIZE_SUCCESS,
  PRIZE_FAIL,
  ENTER_SUCCESS_PRIZE,
  ENTER_SUCCESS_USER
} from "./types";
import setConfigHeader from "../utils/setConfigHeader";

export const loadPrizes = () => async dispatch => {
  try {
    const res = await axios.get("/api/prizes");
    dispatch({
      type: PRIZES_LOADED,
      payload: res.data
    });
  } catch (error) {
    console.log(error);
    dispatch({
      type: CLEAR_PRIZES
    });
  }
};

export const loadSinglePrize = id => async dispatch => {
  try {
    const res = await axios.get(`/api/prizes/${id}`);
    dispatch({
      type: PRIZE_SUCCESS,
      payload: res.data
    });
  } catch (error) {
    dispatch({
      type: PRIZE_FAIL
    });
  }
};

export const enterTickets = (amount, prizeId) => async dispatch => {
  const config = setConfigHeader();
  const body = JSON.stringify({ amount, prizeId });

  try {
    const res = await axios.put("/api/interactive/enterTickets", body, config);
    //  dispatch({
    //    type: ENTER_SUCCESS_PRIZE,
    //    payload: res.data.ticketPool
    //  });
    //  dispatch({
    //    type: ENTER_SUCCESS_USER,
    //    payload: res.data.userTickets
    //  });
    console.log(res.data);
  } catch (error) {
    console.log(error);
  }
};
