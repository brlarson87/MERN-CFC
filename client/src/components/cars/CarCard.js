import React, { Fragment } from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";

import formatNumber from "../../utils/formatNumber";
import userTickets from "../../utils/userTickets";
import progressBar from "../../utils/progressBar";

const CarCard = props => {
  return (
    <Fragment>
      <div className='car-card'>
        <h2 className='car-card__title'>{props.prize.car}</h2>
        <img
          src={props.prize.pictures[0]}
          alt='2020-c8-1'
          className='car-card__image'
        />

        <div className='car-card__bg-overlay'>&nbsp;</div>
        <div className='car-card__details'>
          {props.user ? (
            <div className='car-card__details--left u-color-secondary'>
              <i className='fas fa-ticket-alt'>
                &nbsp; x {userTickets(props.user.tickets, props.prize._id)}
              </i>
              <div className='u-color-primary u-margin-top-sm'>
                <i className='fas fa-ribbon'>
                  &nbsp; x {props.prize.charityPool.length}
                </i>
              </div>
            </div>
          ) : (
            <div className='car-card__details--left u-color-primary u-margin-top-sm'>
              <i className='fas fa-ribbon'>
                &nbsp; x {props.prize.charityPool.length}
              </i>
            </div>
          )}
          <div className='car-card__details--right'>
            <span className='car-card__tickets-in'>
              {formatNumber(props.prize.ticketPool)}
            </span>{" "}
            /
            <span className='car-card__total-tickets'>
              &nbsp;{formatNumber(props.prize.prizeTotal)}&nbsp;Tks
            </span>
          </div>
        </div>
        <div className='car-card__progress-bar'>
          <span
            className='car-card__progress-bar__filler'
            style={{
              width: progressBar(props.prize.ticketPool, props.prize.prizeTotal)
            }}
          ></span>
        </div>
        <Link
          to={`/cardetails/${props.prize._id}`}
          className='btn btn--quick'
          onClick={() => window.scrollTo(0, 0)}
          style={{ marginBottom: "3px" }}
        >
          Quick Enter
        </Link>
        <Link
          to={`/cardetails/${props.prize._id}`}
          className='btn btn--primary u-margin-bottom-sm'
          onClick={() => window.scrollTo(0, 0)}
        >
          Details
        </Link>
      </div>
    </Fragment>
  );
};

const mapStateToProps = state => ({
  user: state.auth.user
});

export default connect(mapStateToProps)(CarCard);
