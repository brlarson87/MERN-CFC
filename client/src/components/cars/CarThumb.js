import React from "react";

const CarThumb = props => {
  return (
    <div
      className='secondary-image'
      onClick={() => {
        props.changeCount(props.keyy);
      }}
    >
      <img
        src={props.pic}
        alt={props.alternate}
        className='secondary-image__img'
      />
    </div>
  );
};

export default CarThumb;
