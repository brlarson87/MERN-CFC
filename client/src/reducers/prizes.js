import {
  PRIZES_LOADED,
  CLEAR_PRIZES,
  PRIZE_SUCCESS,
  PRIZE_FAIL,
  ENTER_SUCCESS_PRIZE,
  PRIZE_PLEDGE_SUCCESS
} from "../actions/types";

const initialState = {
  prizes: [],
  loading: true,
  prize: null
};

export default function(state = initialState, action) {
  const { type, payload } = action;

  switch (type) {
    case PRIZES_LOADED:
      return {
        ...state,
        prizes: payload,
        loading: false
      };
    case PRIZE_SUCCESS:
      return {
        ...state,
        loading: false,
        prize: payload[0]
      };
    case CLEAR_PRIZES:
      return {
        ...state,
        prizes: [],
        loading: false,
        prize: null
      };
    case PRIZE_FAIL:
      return {
        ...state,
        loading: false,
        prize: null
      };
    case ENTER_SUCCESS_PRIZE:
    case PRIZE_PLEDGE_SUCCESS:
      return {
        ...state,
        loading: false,
        prize: payload
      };
    default:
      return state;
  }
}
