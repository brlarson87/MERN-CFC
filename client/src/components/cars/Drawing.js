import React, { Fragment, useEffect, useState } from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";

import Timer from "./Timer";
//import moment from "moment";

import { loadResultAndPrize } from "../../actions/prizes";

//import formatNumber from "../../utils/formatNumber";
//import { formatTime } from "../../utils/formatNumber";
import { formatTicketNumber } from "../../utils/numberPoolsEntered";

import SingleTicket from "../dashboard/SingleTicket";

const Drawing = ({ match, loadResultAndPrize, result, prize }) => {
  useEffect(() => {
    loadResultAndPrize(match.params.id);
  }, [loadResultAndPrize, match.params.id]);

  const [resultData, setResultData] = useState({
    grandWinner: {},
    secondaryWinners: [],
    ticketWinners: [],
    charityWinner: {},
    stopTimer: false,
  });

  const startAnimation = () => {
    setResultData({
      ...resultData,
      stopTimer: true,
    });

    setTimeout(() => {
      setResultData({
        ...resultData,
        stopTimer: true,
      });
    }, 2000);
  };

  return (
    <div className='drawing-container'>
      {result && prize && (
        <Fragment>
          <Link to='/cars' style={{ textDecoration: "none" }}>
            <i className='fas fa-arrow-alt-circle-left drawing-back-arrow'></i>
          </Link>
          <div className='container container--colored'>
            <div className='timer'>
              {/* <div className='timer__time'>{formatTime(time)}</div> */}
              {!resultData.stopTimer && (
                <Fragment>
                  <div className='timer__title'>Winners are picked in</div>
                  <Timer
                    time={result.drawingTime}
                    startAnimation={startAnimation}
                  />
                </Fragment>
              )}
            </div>
            <h1 className='car-title'>{prize.car}</h1>
            <div className='drawing-flex'>
              <div className='drawing-flex__box'>
                <img
                  className='thumbnail__car-image'
                  src={prize.pictures[0]}
                  alt='Car'
                />
              </div>

              <div className='drawing-flex__box'>
                <h2 className='drawing-flex__box__title'>Secondary Prize</h2>
                <h3 className='drawing-flex__box__value'>
                  {prize.secondaryPrizes.prize}
                </h3>
              </div>

              <div className='drawing-flex__box'>
                <h2 className='drawing-flex__box__title'>Ticket Prize</h2>
                <h3>
                  <i className='fas fa-ticket-alt u-color-secondary'>
                    &nbsp; x {prize.ticketsPrize.amount}
                  </i>
                </h3>
              </div>

              <div className='drawing-flex__box'>
                <h2 className='drawing-flex__box__title'>Charity Donation</h2>
                <h3 className='drawing-flex__box__value'>
                  ${prize.charityAmount}
                </h3>
              </div>
            </div>

            <div className='winning-numbers-flex'>
              <div className='winner-box'>
                <h2 className='result-title result-title--gold'>
                  Grand Prize Winner
                </h2>
                {resultData.stopTimer && (
                  <div className='ticket-container ticket-container--single'>
                    <div className='ticket-container__single-ticket ticket-container__single-ticket--single'>
                      <i className='fas fa-ticket-alt'>
                        <span className='ticket-container__single-ticket--number ticket-container__single-ticket--number--single'>
                          {formatTicketNumber(
                            result.grandPrizeTicketNumber,
                            prize.prizeTotal
                          )}
                        </span>
                      </i>
                    </div>
                  </div>
                )}
              </div>

              <div className='winner-box'>
                <h2 className='result-title'>Charity Winner</h2>
                <div className='ticket-container ticket-container--single'></div>
              </div>
            </div>

            <div className='winning-numbers-flex winning-numbers-flex--lg'>
              <div className='item secondary-winner-box'>
                <h2 className='result-title result-title--secondary'>
                  Secondary Winners
                </h2>
                <div className='ticket-container ticket-container--secondary'>
                  {/* Map Secondary Tickets into SingleTicket Component */}
                  {resultData.stopTimer &&
                    result.secondaryPrizes.map((num, index) => (
                      <SingleTicket
                        number={num}
                        total={prize.prizeTotal}
                        key={index}
                        dash={false}
                      />
                    ))}
                </div>
              </div>
              <div className='item ticket-winner-box'>
                <h2 className='result-title result-title--secondary'>
                  Ticket Winners
                </h2>
                <div className='ticket-container ticket-container--secondary'>
                  {/* Map ticketPrize Tickets into SingleTicket Component */}
                  {resultData.stopTimer &&
                    result.ticketPrizes.map((num, index) => (
                      <SingleTicket
                        number={num}
                        total={prize.prizeTotal}
                        key={index}
                        dash={false}
                      />
                    ))}
                </div>
              </div>
            </div>
          </div>
        </Fragment>
      )}
    </div>
  );
};

const mapStateToProps = (state) => ({
  result: state.results.result,
  prize: state.prizes.prize,
});

export default connect(mapStateToProps, {
  loadResultAndPrize,
})(Drawing);
