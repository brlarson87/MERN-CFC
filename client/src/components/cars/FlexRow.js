//CORE REACT
import React, { Fragment } from "react";

//Components
import CarCard from "./CarCard";

const FlexRow = ({ chunk, user }) => {
  return (
    <Fragment>
      <div className='flex-row'>
        {/*----------CARCARD COMPONENTS-----------*/}
        {user
          ? chunk.map((prize) => (
              <CarCard prize={prize} key={prize._id} user={user} />
            ))
          : chunk.map((prize) => <CarCard prize={prize} key={prize._id} />)}
      </div>
    </Fragment>
  );
};

export default FlexRow;
