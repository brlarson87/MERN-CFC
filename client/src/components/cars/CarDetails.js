//CORE REACT
import React, { useEffect, useState, Fragment } from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";

//Utils
import formatNumber from "../../utils/formatNumber";
import ticketsEnteredInPool from "../../utils/ticketsEnteredInPool";
import ticketStatus from "../../utils/ticketStatus";

//Actions
import { showConfirmModal } from "../../actions/modal";
import { loadSinglePrize } from "../../actions/prizes";

//COMPONENTS
import EnterTickets from "../auth/EnterTickets";
import CarThumb from "./CarThumb";
//import Spinner from "../layout/Spinner";

const CarDetails = ({
  loadSinglePrize,
  match,
  prize,
  loading,
  user,
  showConfirmModal,
}) => {
  useEffect(() => {
    loadSinglePrize(match.params.id);
  }, [loadSinglePrize, match.params.id]);

  const [count, setCount] = useState(0);

  const addOne = () => {
    if (count === prize.pictures.length - 1) {
      setCount(0);
    } else {
      setCount(count + 1);
    }
  };

  const subtractOne = () => {
    if (count === 0) {
      setCount(prize.pictures.length - 1);
    } else {
      setCount(count - 1);
    }
  };

  const changeCount = (x) => {
    setCount(x);
  };

  return (
    <div className='details-container'>
      <div className='left-col'>
        <div className='details'>
          <h2 className='details__heading--1'>
            {!loading && prize && prize.car}
          </h2>
          <div className='details__group'>
            <span className='details__group--left'>Total Tickets</span>
            <span className='details__group--right'>
              {!loading && prize && formatNumber(prize.prizeTotal)}
            </span>
          </div>
          <div className='details__group'>
            <span className='details__group--left'>Tickets Entered</span>
            <span className='details__group--right'>
              {!loading && prize && formatNumber(prize.ticketPool)}
            </span>
          </div>

          {user && (
            <div className='details__group'>
              <span className='details__group--left'>Your Tickets</span>
              <span className='details__group--right'>
                {!loading && user && prize
                  ? ticketsEnteredInPool(user.tickets, match.params.id)
                  : 0}
              </span>
            </div>
          )}

          <div className='details__group'>
            <span className='details__group--left'>Charity Donation</span>
            <span className='details__group--right'>
              {!loading && prize && formatNumber(prize.charityAmount, true)}
            </span>
          </div>

          <h4 className='details__heading--2'>Secondary prizes</h4>

          <div className='details__group'>
            <span className='details__group--left'>
              Places 2 - {!loading && prize && prize.secondaryPrizes.places + 1}
            </span>
            <span className='details__group--right'>
              {!loading && prize && prize.secondaryPrizes.prize}
            </span>
          </div>
          <div className='details__group'>
            <span className='details__group--left'>
              Places {!loading && prize && prize.secondaryPrizes.places + 2} -{" "}
              {!loading &&
                prize &&
                prize.secondaryPrizes.places + 1 + prize.ticketsPrize.places}
            </span>
            <span className='details__group--right u-color-secondary'>
              <i className='fas fa-ticket-alt'></i> x{"  "}
              {!loading && prize && prize.ticketsPrize.amount}
            </span>
          </div>

          <h4 className='details__heading--2'>Features</h4>
          <div className='details__features'>
            <p className='details__features--text'>
              {!loading && prize && prize.carDetails}
            </p>
            <a href='/' className='btn--block'>
              Full Features
            </a>
          </div>

          {/*****************************************/}
          {/*---------ENTERTICKETS COMPONENT---------*/}
          {/*****************************************/}
          {prize && user ? (
            <EnterTickets
              id={prize._id}
              ticketTotal={prize.ticketPool}
              showConfirmModal={showConfirmModal}
              activeUserTickets={ticketStatus(user.tickets).active}
              prizeName={prize.car}
              thumbnail={prize.pictures[0]}
              prizeTotal={prize.prizeTotal}
            />
          ) : (
            <Fragment>
              <h5 className='details__ticket-form--title u-margin-top-lg'>
                Login to enter
              </h5>
              <Link to='/login' className='btn--block u-margin-top-sm'>
                Login
              </Link>
            </Fragment>
          )}
        </div>
      </div>

      <div className='right-col'>
        <div className='images-container'>
          {/*----------MAIN IMAGE-----------*/}
          <div className='images-container__top'>
            <img
              src={prize && prize.pictures[count]}
              alt='car'
              className='main-image'
            />
            <div className='arrow arrow-left' onClick={() => subtractOne()}>
              <i className='fas fa-arrow-left'></i>
            </div>
            <div className='arrow arrow-right' onClick={() => addOne()}>
              <i className='fas fa-arrow-right'></i>
            </div>
          </div>

          {/*****************************************/}
          {/*----------CARTHUMB COMPONENT-----------*/}
          {/*****************************************/}
          <div className='images-container__bottom'>
            {/*----- Loop through prize pictures ---------*/}
            {prize &&
              prize.pictures.map((pic, i) => (
                <CarThumb
                  pic={pic}
                  keyy={i}
                  key={i}
                  alternate={prize.car}
                  changeCount={changeCount}
                />
              ))}
          </div>
        </div>
      </div>
    </div>
  );
};

const mapStateToProps = (state) => ({
  prize: state.prizes.prize,
  loading: state.auth.loading,
  user: state.auth.user,
});

export default connect(mapStateToProps, {
  loadSinglePrize,
  showConfirmModal,
})(CarDetails);
