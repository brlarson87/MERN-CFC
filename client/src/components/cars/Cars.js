import React, { useEffect, Fragment } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";

//Components
import FlexRow from "./FlexRow";
import Spinner from "../layout/Spinner";
import Footer from "../layout/Footer";
import Header from "../layout/Header";
import TicketConfirm from "../modals/TicketConfirm";
//Actions
import { loadPrizes } from "../../actions/prizes";
import { setAlert } from "../../actions/alert";
import { showConfirmModal } from "../../actions/modal";
//Utilities
import chunks from "../../utils/chunks";

export const Cars = ({
  loadPrizes,
  prizes,
  loading,
  user,
  showConfirmModal,
  setAlert,
}) => {
  useEffect(() => {
    loadPrizes();
  }, [loadPrizes]);
  return (
    <Fragment>
      <Header />
      <div className='container'>
        <TicketConfirm />
        {/*----------FLEXROW COMPONENT-----------*/}
        {!loading && prizes ? (
          prizes.map((chunk, index) => (
            <FlexRow
              chunk={chunk}
              key={index}
              user={user}
              showConfirmModal={showConfirmModal}
              setAlert={setAlert}
            />
          ))
        ) : (
          <Spinner />
        )}{" "}
      </div>
      {!loading && <Footer />}
    </Fragment>
  );
};

Cars.propTypes = {
  loadPrizes: PropTypes.func,
  prizes: PropTypes.array,
  loading: PropTypes.bool,
};

const mapStateToProps = (state) => ({
  prizes: chunks(state.prizes.prizes),
  loading: state.prizes.loading,
  user: state.auth.user,
});

export default connect(mapStateToProps, {
  loadPrizes,
  showConfirmModal,
  setAlert,
})(Cars);
