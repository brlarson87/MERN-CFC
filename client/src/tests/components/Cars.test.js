import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import { Cars } from "../../components/cars/Cars";

import prizes from "../fixtures/prizes";

describe("CARS COMPONENT", () => {
  it("renders Cars Component successfully without user", () => {
    const wrapper = shallow(<Cars prizes={prizes} />);

    expect(wrapper.find(FlexRow)).toHaveLength(4);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
