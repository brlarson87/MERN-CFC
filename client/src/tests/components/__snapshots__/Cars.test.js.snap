// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CARS COMPONENT renders Cars Component successfully without user 1`] = `
<Fragment>
  <Connect(Header) />
  <div
    className="container"
  >
    <Connect(TicketConfirm) />
    <FlexRow
      chunk={
        Object {
          "_id": "1",
          "active": true,
          "car": "Audi R8",
          "carDetails": "The 2020 V10 R8 is a real wheel drive coupe putting out 562 horspower.",
          "charityAmount": 20000,
          "charityPledgeAmount": 5,
          "charityPool": Array [
            Object {
              "charityId": "1",
              "name": "SecondHand Hounds",
              "prizeId": "1",
            },
          ],
          "dateAdded": "2020-01-30T02:15:09.284Z",
          "pictures": Array [
            "https://i.ibb.co/qrfRhvY/r8-5.jpg",
            "https://i.ibb.co/D5VhSjy/r8-1.jpg",
            "https://i.ibb.co/4Nh17yP/r8-2.jpg",
          ],
          "prizeTotal": 100,
          "secondaryPrizes": Object {
            "places": 75,
            "prize": "Apple Airpods",
          },
          "ticketPool": 5,
          "ticketsPrize": Object {
            "amount": 15,
            "places": 300,
          },
        }
      }
      key="0"
    />
    <FlexRow
      chunk={
        Object {
          "_id": "2",
          "active": true,
          "car": "2019 Camaro Valkyrja",
          "carDetails": "This restomod camaro from Ringbrothers is a masterpiece putting out 890 horsepower. The amount of detail put into this car is incredible.",
          "charityAmount": 30000,
          "charityPledgeAmount": 10,
          "charityPool": Array [],
          "dateAdded": "2020-01-30T02:21:06.224Z",
          "pictures": Array [
            "https://i.ibb.co/j6mx5C8/cv-13.jpg",
            "https://i.ibb.co/swz5f4W/cv-16.jpg",
            "https://i.ibb.co/FKt9vt0/cv-15.jpg",
          ],
          "prizeTotal": 200,
          "secondaryPrizes": Object {
            "places": 5,
            "prize": "13.5in macbook pro",
          },
          "ticketPool": 2,
          "ticketsPrize": Object {
            "amount": 350,
            "places": 15,
          },
        }
      }
      key="1"
    />
    <FlexRow
      chunk={
        Object {
          "_id": "3",
          "active": true,
          "car": "2020 Mercedes AMG GT C",
          "carDetails": "Powered by a 4.0L AMG V8 Biturbo engine making 550 horsepower, this car sounds as good as it looks.",
          "charityAmount": 17500,
          "charityPledgeAmount": 5,
          "charityPool": Array [],
          "dateAdded": "2020-01-30T02:21:34.161Z",
          "pictures": Array [
            "https://i.ibb.co/3h3WN2y/amg-1.jpg",
            "https://i.ibb.co/fpGC1nP/amg-2.jpg",
            "https://i.ibb.co/SKKgThZ/amg-3.jpg",
          ],
          "prizeTotal": 110,
          "secondaryPrizes": Object {
            "places": 10,
            "prize": "60in 5k TV",
          },
          "ticketPool": 5,
          "ticketsPrize": Object {
            "amount": 150,
            "places": 30,
          },
        }
      }
      key="2"
    />
    <FlexRow
      chunk={
        Object {
          "_id": "4",
          "active": true,
          "car": "Hellcat Redeye Widebody",
          "carDetails": "The Challenger Hellcat Widebody automatic hits 60 mph in 3.8 seconds and storms through the quarter in 11.8 seconds at 126 mph",
          "charityAmount": 7500,
          "charityPledgeAmount": 2,
          "charityPool": Array [],
          "dateAdded": "2020-01-30T02:48:25.004Z",
          "pictures": Array [
            "https://i.ibb.co/92mW0Qg/hellcat-12.jpg",
            "https://i.ibb.co/Hhx9PPF/hellcat-14.jpg",
            "https://i.ibb.co/m0CR204/hellcat-15.jpg",
          ],
          "prizeTotal": 80,
          "secondaryPrizes": Object {
            "places": 5,
            "prize": "$200 Best Buy card",
          },
          "ticketPool": 8,
          "ticketsPrize": Object {
            "amount": 50,
            "places": 10,
          },
        }
      }
      key="3"
    />
     
  </div>
  <Footer />
</Fragment>
`;
